prov/gni/src/gnix_av.c

```diff
@@ -172,6 +173,10 @@ static int table_insert(struct gnix_fid_av *av_priv, const void *addr,
 		/* check if this ep_name fits in the av context bits */
 		if (temp->name_type & GNIX_EPN_TYPE_SEP) {
 			if ((1 << av_priv->rx_ctx_bits) < temp->rx_ctx_cnt) {
+				if (flags && FI_SYNC_ERR) {
+					entry_err[i] = -FI_EINVAL;
+					fi_addr[i] = FI_ADDR_NOTAVAIL;
+				}
```

Comment:
The man page says that "Providers are allowed to abort insertion operations in the case of an error." 
After re-reading the man page, it also states "Addresses that are not inserted because they were aborted will fail with an error code of FI_ECANCELED."
I'm only setting the one before returning. I think it will be more straightforward to finish trying to insert all and then return. I'll clean this up.